tests/monitoring/metrics.go

```diff
@@ -0,0 +1,121 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package monitoring
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/onsi/gomega/types"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	virtctlpause "kubevirt.io/kubevirt/pkg/virtctl/pause"
+
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/clientcmd"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+var _ = Describe("[Serial][sig-monitoring]Prometheus Metrics", Serial, decorators.SigMonitoring, func() {
+	var err error
+	var virtClient kubecli.KubevirtClient
+
+	BeforeEach(func() {
+		virtClient = kubevirt.Client()
+	})
+
+	Context("VM Metrics", func() {
+		var vm *v1.VirtualMachine
+
+		BeforeEach(func() {
+			vmi := tests.NewRandomVMI()
+			vm = tests.NewRandomVirtualMachine(vmi, false)
+
+			By("Create a VirtualMachine")
+			_, err = virtClient.VirtualMachine(vm.Namespace).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		AfterEach(func() {
+			By("Deleting the VM")
+			Expect(virtClient.VirtualMachine(vm.Namespace).Delete(context.Background(), vm.Name, &metav1.DeleteOptions{})).To(Succeed())
+		})
+
+		checkMetricTo := func(metric string, labels map[string]string, matcher types.GomegaMatcher) {
+			Eventually(func() int {
+				v, err := getMetricValueWithLabels(virtClient, metric, labels)
+				if err != nil {
+					return -1
+				}
+				i, err := strconv.Atoi(v)
+				Expect(err).ToNot(HaveOccurred())
+				return i
+			}, 3*time.Minute, 20*time.Second).Should(matcher)
+		}
+
+		It("should be available for a running VM", func() {
+			By("Start the VM")
+			vm = tests.StartVirtualMachine(vm)
+
+			By("Checking that the VM metrics are available")
+			checkMetricTo(
+				"kubevirt_vmi_cpu_usage_seconds",
+				map[string]string{"name": vm.Name, "namespace": vm.Namespace},
+				BeNumerically(">=", 0),
+			)
+		})
```

Comment:
IMHO we don't need this test since we are already performing this test in https://github.com/kubevirt/kubevirt/blob/4c70bae4903a5396b771395b36f007015b25d32f/tests/monitoring/monitoring.go#L420

Reply:
removed the duplication